### PR TITLE
RestToCode: first pull

### PR DESCRIPTION
**Pull request by acmitchell11**
This specific version is so that you can look at and test the code since my pytest is doing weird things. Do not actually merge yet because this is unfinished. I tried installing the 64 bit version of python (uninstalled 32 bit) and pytest worked one time. It then resumed giving me the same errors. They are OS errors, and I just installed the newest Windows update. This might be the cause of the new errors. Also, on the test that actually worked, I tested some of the files in CodeChat to insure round trip stability of the file interpreters. CodeChat.css was interpreted by code_to_rest_file as 'CSS+Lasso' as the correct lexer. This made the test fail with a Key Error because we do not support 'CSS+Lasso'. I have no idea why it was allowed to suggest this lexer, and it might have been a one time error. Replicating this error might be helpful in finding a bug in the code.

### CODE REVIEW

**Comment by acmitchell11** [comment]
I also got Pytest to stop throwing the OS Windows 87 error. It turns out that I just needed to uninstall Python 3.6 and use 3.5 instead. I don't know why, but I think that one of the dependencies has trouble operating in 3.6

**Comment by bjones1** [comment]
Awesome. Nice job debugging. Not sure why.

**Review by bjones1 — CHANGES_REQUESTED** [review]

File: CodeChat/CodeToRest.py [hunk 1]
```diff
@@ -202,6 +202,8 @@ def get_lexer(
   # See options_.
   **options):
 
+    options.setdefault('tabsize',4)
```
Comment (acmitchell11) [inline CodeChat/CodeToRest.py]: Done

File: CodeChat/CommentDelimiterInfo.py [hunk 2]
```diff
@@ -105,12 +105,11 @@
   # PODs begin and end on a line starting with =string. Toss the entire line by
   # making the delimiter length large. This allows it to work with ``=head1``,
   # ``=begin``, ``=begin comment``, and a host of others.
-  ##                  #, =a string, =another string
-  'Perl':           ( '#',   '=begin ' + '*'*1000, '=end ' + '*'*1000),
+  # PODs not supported. Only single line comments
```
Comment (acmitchell11) [inline CodeChat/CommentDelimiterInfo.py]: Done

File: CodeChat/RestToCode.py [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+# ====================
+# This section covers all functions that support the main two functions rest_to_code_string_
+# and rest_to_code_file_. 
+# |
+#
+# _`find_file_ext`: Find the file extension needed, given the language name
+def find_file_ext(
+  # See lang_.
+  lang):
+    # TODO Make a test for finding the file extention.
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Done

File: CodeChat/RestToCode.py [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+    # Else, return ``None``
+    # TODO do a for/else statement. use break to
+    if found == 1:
+        # Use only the ``.py`` part of ``*.py`` and similar.
+        file_ext_list = file_ext_raw.split('*', 1)
+        file_ext = file_ext_list[-1]
+        return file_ext
+    else:
+        return None'''
+# |
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Done

File: CodeChat/RestToCode.py [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+#
+# _`file_info`: Gather the location of the file to be translated and the name of the language
+# that the file will be translated into.
+# This is the only part of the code that needs to be filled out by the user at runtime.
+#
+# The user can skip this by providing a language in the parameters of rest_to_code_file_.
+def file_info(
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Taken out

File: CodeChat/RestToCode.py [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+  # are `enough lines <number_consecutive_comments>`_ to be considered a block comment.
+  line_counter):
+    # Grab the comment delimiters for the given language
+    comment_delimiters = COMMENT_DELIMITER_INFO[lang]
+    # Check to see what kinds of comments the language supports
+    comment_type = language_comment_type(comment_delimiters)
+    # Create an inline comment
+    if is_block_comment is False and comment_type[0] is True:
+        return formulate_inline_comment(line, comment_delimiters)
+    elif is_block_comment is True and comment_type[1] is False:
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Done

File: CodeChat/RestToCode.py [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+        return formulate_inline_comment(line, comment_delimiters) # TODO create a test to check if this is right... I think it might only do the first line of the block comment and skip the rest.
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Done

File: CodeChat/RestToCode.py [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+    # Create a block comment
+    else:
+        if is_block_comment is False:
+            line_counter = None
+        return formulate_block_comment(line, comment_delimiters, position, line_counter)
+# |
+#
+# _`formulate_inline_comment`: Formulates an inline comment
+def formulate_inline_comment(
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Done

File: CodeChat/RestToCode.py [hunk 9]
```diff
@@ -0,0 +1,432 @@
...
+  # See line_.
+  line,
+  # See comment_delimiters_.
+  comment_delimiters):
+
+    f = '{} '.format(comment_delimiters[0]) + line + '\n'
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Done

File: CodeChat/RestToCode.py [hunk 10]
```diff
@@ -0,0 +1,432 @@
...
+  # See line_counter_.
+  line_counter):
+
+    # This covers the case that the language does not support inline comments.
+    # It places block comment delimiters around a single line to give an inline effect
+    # There is no added space between the end of the comment and the end delimiter because this
+    # space is not taken out in the Code to reST translation. If a space is added, it is no 
+    # longer round trip stable. (It adds a space every time it is translated.)
+    if line_counter is None:
+        f = '{} '.format(comment_delimiters[1]) + line + '{}'.format(comment_delimiters[2]) + '\n'
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Done

File: CodeChat/RestToCode.py [hunk 11]
```diff
@@ -0,0 +1,432 @@
...
+# ==============
+# This section contains the main two functions rest_to_code_string_
+# and rest_to_code_file_.
+# |
+#
+# _`rest_to_code_file`: This function uses rest_to_code_string_ to convert a reST file into
+# another language. Inputs a reST file, outputs a code file.
+def rest_to_code_file(
+  # _`source_rst_path`: Path to a source reST file to process.
+  source_rst_path=None,
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Got rid of the default value

File: CodeChat/RestToCode.py [hunk 12]
```diff
@@ -0,0 +1,432 @@
...
+  # _`out_path`:Path to a destination code file to create. It will be overwritten if it
+  # already exists.
+  out_path=None,
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Added the comment and code for this to work

File: CodeChat/RestToCode.py [hunk 13]
```diff
@@ -0,0 +1,432 @@
...
+  # _`input_encoding`: Encoding to use for the input file. The default of None detects the encoding
+  # of the input file.
+  input_encoding=None,
+  # _`output_encoding`: Encoding to use for the output file.
+  output_encoding='utf-8'):
+
+    if lang is None:
+        file_name, lang, file_ext = file_info(source_rst_path, out_path)
+
+        if source_rst_path is None:
```
Comment (acmitchell11) [inline CodeChat/RestToCode.py]: Changed this part of the code entirely, so this is fixed

**Comment by acmitchell11** [comment]
All requested changes are completed and uploaded. You should be able to see them.

**Pull request merged**